src/NUnitFramework/framework/Constraints/IndexerConstraint.cs

```diff
@@ -0,0 +1,84 @@
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework.Internal;
+
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// IndexerConstraint extracts a named property and uses
+    /// its value as the actual value for a chained constraint.
+    /// </summary>
+    public class IndexerConstraint : PrefixConstraint
+    {
+        private readonly Type[] _argumentTypes;
+        private readonly object[] _arguments;
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IndexerConstraint"/> class.
+        /// </summary>
+        /// <param name="indexerArguments">The argument list for the indexer.</param>
+        /// <param name="baseConstraint">The constraint to apply to the indexer.</param>
+        public IndexerConstraint(IEnumerable<object> indexerArguments, IConstraint baseConstraint)
+            : base(baseConstraint)
+        {
+            _arguments = indexerArguments.ToArray();
+            _argumentTypes = _arguments.Select(a => a.GetType()).ToArray();
+
+            DescriptionPrefix = $"indexer {StringifyArguments()}";
+        }
+
+        /// <summary>
+        /// Test whether the constraint is satisfied by a given value
+        /// </summary>
+        /// <param name="actual">The value to be tested</param>
+        public override ConstraintResult ApplyTo<TActual>(TActual actual)
+        {
+            Guard.ArgumentNotNull(actual, nameof(actual));
+
+            if (!(actual is Type actualType))
+            {
+                actualType = actual.GetType();
+            }
+
+            var indexerInfo = Reflect.GetIndexerByName(actualType, _argumentTypes);
+            if (indexerInfo?.GetGetMethod() == null)
+            {
+                throw new ArgumentException($"Indexer with {StringifyArguments()} was not found on {actualType}.");
+            }
```

Comment:
Applying the same logic:

```suggestion
            var getter = Reflect.GetIndexerByName(actualType, _argumentTypes)?.GetGetMethod()
                ?? throw new ArgumentException($"Indexer with {StringifyArguments()} was not found on {actualType}.");
```

(same in the other file, minus the `throw` part)